README.md

```diff
@@ -31,62 +30,43 @@ curl -X POST localhost:3000 -H 'content-type: application/json' -d '{"hello": "w
-    * `opts.cores.methods?array(string)`: an array of methods to accept besides the default. Default: `PUT`, `POST`, `DELETE`, `GET`, `OPTIONS`
-
-### `Five#use(middleware:function)`
-Add a new middleware to the stack.  Middleware will be processed in the order in
-which they are added, which means they will be run after the built-in middleware.
-
-* `middleware(request:object, response:object, next:function):function` -You must either call `next` or send data to the client when you are finshed.
-
-Since the middleware signature is the same express/restify, you *might* be able to use existing middleware with take-five, but ymmv.
-
-### `Five#router(namespace:string):object`
-Namespace routes. All routes defined off this router will be prefixed with the supplied
-namespace.  The methods have the same signature as the router provided.
+* `opts.allowHeaders?:array(string)`: an array of headers to accept besides the default. Default: `Content-Type`, `Accept`, `X-Requested-With`
+* `opts.allowOrigin?:string`: What origin(s) are accepted. Deafult: `*`
+* `opts.allowCredentials?:boolean`: Allow or deny credentials. Default: `true`
+* `opts.allowMethods?array(string)`: an array of methods to accept besides the default. Default: `PUT`, `POST`, `DELETE`, `GET`, `OPTIONS`, `PATCH`
+* `opts.methods?array(string)`: An array of methods to create route handlers for. Default: `PUT`, `POST`, `DELETE`, `GET`, `OPTIONS`, `PATCH`
+* `opts.http?object`: options for `http(s).createServer`. If you supply `key`,
+    `cert` and `ca` as options, it will assume you are trying to create an https server`
 
 ### `Five#get(route:string, handler:(function|array(function)))`
 ### `Five#post(route:string, handler:(function|array(function)))`
 ### `Five#put(route:string, handler:(function|array(function)))`
+### `Five#patch(route:string, handler:(function|array(function)))`
 ### `Five#delete(route:string, handler:(function|array(function)))`
 Add a new route to the server. Routes may be added after the server has been
 started. You can supply either a single function or an array of functions to call.
 The array will be traversed in the order it is supplied
 
 * `route:string` A [wayfarer](https://github.com/yoshuawuyts/wayfarer) route definition.
-* `handler(request:object, response:object, next:function):function`: The handler for this route.
-
-Since this is an augmented instance of [http.Server](https://nodejs.org/api/http.html#http_class_http_server)
-all methods and properties are available on this as well.
-
-### `request:object`
-The [`http.IncomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage)
-object extended with:
+* `handler(request:object, response:object, ctx:object):function`: The handler for this route.
 
+#### `ctx:object`
 * `params?:object`: query parameters from the URL (if any)
 * `urlParams?:object`: Named parameters from the route definition as provided by wayfarer
```

Comment:
As long as this is a major bump, renaming `params` to "query" and `urlParams` to "params" makes a lot more sense to me.

Reply:
Agree!

Reply:
This is there now!